stellar-dotnet-sdk/TransactionBuilder.cs

```diff
@@ -0,0 +1,160 @@
+        ///     See: https://www.stellar.org/developers/learn/concepts/list-of-operations.html
+        /// </summary>
+        /// <param name="operation">operation</param>
+        /// <returns>Builder object so you can chain methods.</returns>
+        public TransactionBuilder AddOperation(Operation operation)
+        {
+            if (operation == null)
+                throw new ArgumentNullException(nameof(operation), "operation cannot be null");
+
+            _operations.Add(operation);
+            return this;
+        }
+
+        /// <summary>
+        ///     Adds a memo to this transaction.
+        ///     See: https://www.stellar.org/developers/learn/concepts/transactions.html
+        /// </summary>
+        /// <param name="memo">Memo</param>
+        /// <returns>Builder object so you can chain methods.</returns>
+        public TransactionBuilder AddMemo(Memo memo)
+        {
+            if (_memo != null)
+                throw new ArgumentException("Memo has been already added.", nameof(memo));
+
+            _memo = memo ?? throw new ArgumentNullException(nameof(memo), "memo cannot be null");
+
+            return this;
+        }
+
+        /// <summary>
+        ///     Adds a time-bounds to this transaction.
+        ///     See: https://www.stellar.org/developers/learn/concepts/transactions.html
+        /// </summary>
+        /// <param name="timeBounds">timeBounds</param>
+        /// <returns>Builder object so you can chain methods.</returns>
+        public TransactionBuilder AddTimeBounds(TimeBounds timeBounds)
+        {
+            if (_timeBounds != null)
+                throw new ArgumentException("TimeBounds has been already added.", nameof(timeBounds));
+
+            _timeBounds = timeBounds ??
+                          throw new ArgumentNullException(nameof(timeBounds), "timeBounds cannot be null");
+
+            return this;
+        }
+
+        /// <summary>
+        ///     Set the transaction fee (in Stroops) per operation.
+        ///     See: https://www.stellar.org/developers/learn/concepts/transactions.html
+        /// </summary>
+        /// <param name="fee">fee (in Stroops) for each operation in the transaction</param>
+        /// <returns>Builder object so you can chain methods.</returns>
+        public TransactionBuilder SetFee(uint fee)
+        {
+            if (_fee <= 0)
+                throw new ArgumentException("Fee must be a positive amount", nameof(fee));
+
+            _fee = fee;
+
+            return this;
+        }
+
+        /// <summary>
+        ///     Builds a transaction. It will increment sequence number of the source account.
+        /// </summary>
+        /// <returns></returns>
+        public Transaction Build()
+        {
+            var operations = _operations.ToArray();
+
+            var totalFee = operations.Length * _fee;
+            if (totalFee > UInt32.MaxValue) throw new InvalidOperationException("Transaction fee overflow");
```

Comment:
I don't think this will detect an overflow since I assume `totalFee ` is a uint32. I think to properly detect an overflow, `totalFee` needs to be a `ulong` and then you can compare against `UInt32.MaxValue`

Reply:
`totalFee` is a long. I will rewrite this to be more clear.